File: 0049-group-anagrams/0049-group-anagrams.cpp
// #include <map>

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        map<string, vector<string>> m;
        for (auto origin: strs) {
            string sorts = origin;
            sort(sorts.begin(), sorts.end());
            if (m.contains(sorts)) {
                m[sorts].push_back(origin);
            } else {
                m[sorts] = {origin};
            }
        }
        for (const auto& pair: m) {
            result.push_back(pair.second);
        }
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        for s in strs:
            sorted_str = ''.join(sorted(s))
            if sorted_str in dic:
                dic[sorted_str].append(s)
            else:
                dic[sorted_str] = [s]
        
        return [arr for k, arr in dic.items()]